File: Lista03/Lista03.R
library(wooldridge)

htv = wooldridge::htv
head(htv)

# Modelo errado abil nao esta fixo
fit = lm(educ ~ motheduc + fatheduc + abil + I(abil^2), htv)
fit$coefficients #coef(lm(educ ~ motheduc + fatheduc + abil + I(abil^2), htv))
# se abil aumenta 1, educ aumenta 0.4015, mantendo os outros fatores fixos

# Para uma analise correta, devemos derivar o modelo com relacao ao abil assim temos que
fit$coefficients[4] + 2 * fit$coefficients[5]

# Ponto de inflexao
- fit$coefficients[5] / (2 * fit$coefficients[4])

File: Lista03/Lista03.jl
using WooldridgeDatasets, DataFrames, GLM

df = DataFrame(wooldridge("HTV"));

modelo = lm(@formula(educ~motheduc + fatheduc + abil + abil^2), df)

parametros = coef(modelo)

B3 = parametros[4]
B4 = parametros[5]

Delta_y = B3 + 2*B4

ponto_inf = -B4 / (2*B3)

#mantendo as outras variaveis constantes, de abil = 0 para abil = 1, educ aumenta B3
#mantendo as outras variaveis constantes, de abil = 1 para abil = x, educ aumenta (Delta_y)*(x-1)
#mantendo as outras variaveis constantes, motheduc aumentando uma unidade, educ aumenta B1
#mantendo as outras variaveis constantes, fatheduc aumentando uma unidade, educ aumenta B2
